test/jdk/java/net/httpclient/DebugLoggerTest.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.lang.ref.Reference;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Predicate;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import jdk.internal.net.http.common.Utils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/*
+ * @test
+ * @summary Verify the behaviour of the debug logger.
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.httpclient.test.lib.common.HttpServerAdapters jdk.test.lib.net.SimpleSSLContext
+ *        DebugLoggerTest
+ * @run main/othervm  DebugLoggerTest
+ * @run main/othervm -Djdk.internal.httpclient.debug=errr  DebugLoggerTest
+ * @run main/othervm -Djdk.internal.httpclient.debug=err  DebugLoggerTest ERR
+ * @run main/othervm -Djdk.internal.httpclient.debug=out  DebugLoggerTest OUT
+ * @run main/othervm -Djdk.internal.httpclient.debug=log  DebugLoggerTest LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true DebugLoggerTest ERR LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,OUT  DebugLoggerTest ERR OUT
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,out,log DebugLoggerTest ERR OUT LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true,log DebugLoggerTest ERR LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=true,out DebugLoggerTest ERR OUT LOG
+ * @run main/othervm -Djdk.internal.httpclient.debug=err,OUT,foo  DebugLoggerTest ERR OUT
```

Comment:
Given that we launch 11 `@run`s in this test, I was curious how long the entire test run would take to complete. Timeout wasn't a concern (I didn't think there would be one), but I was curious if marking them as multiple `@test` runs to run them parallely would gain any performance improvements. But after having run this test locally in the current form, I see that the whole test completes in slightly over 2 seconds. So there's no point in thinking of any optimizations.